### ISSUE TITLE
QuotedEntityChecker.getLabel(IRI) returns values from axiom annotations

### ISSUE THREAD

**Issue by beckyjackson** [ontodev/robot#589]
If you have a class that has an annotations, e.g.:
![image](https://user-images.githubusercontent.com/16750150/68704436-292c3e80-0541-11ea-89e6-d5ef156dd419.png)
![image](https://user-images.githubusercontent.com/16750150/68704184-c63aa780-0540-11ea-9e6c-07718df58e9b.png)

The `rdfs:label` in the annotated axiom may be returned as the label from the QuotedEntityChecker. It looks like:
```
EntitySearcher.getAnnotations(entity, ontologies, property);
```
... returns the annotations on axioms as well, as long as they lead back to the provided entity. The annotations lose the subject, so you don't know if it's on the class or on another annotation. The example above would have two values from `rdfs:label`: '3,5-dibromo-L-tyrosine' and 'Gmelin'. This means 'Gmelin' could incorrectly show up as the single label picked.

You can get the annotation assertion axioms for an entity, but it will return *all* the annotation axioms for that entity. Then you must filter out the ones that use the correct property and are not for an anonymous subject. In theory this would work, but it takes way too long for bigger ontologies.

@jamesaoverton @cmungall - do you have any other ideas on how we could efficiently get *only* `rdfs:label` annotations on an entity?

**Comment by cmungall** [comment]
I never use EntitySearcher, sorry. I feel I have written lots of verbose
code to get labels for classes using the owlapi many times using different
approaches, but have forgotten them all

**Comment by matentzn** [comment]
Maybe @ignazio1977 can help?

**Comment by ignazio1977** [comment]
I've created a pull request - I think your code needs to use getAnnotationObjects.

**Comment by matentzn** [comment]
Thanks for the help Ignazio!

**Comment by beckyjackson** [comment]
Resolved by 06e7855
